convert_hf_to_gguf.py

```diff
@@ -5065,13 +5049,20 @@ def set_vocab(self):
         special_vocab._set_special_token("eot", tokenizer.get_added_vocab()["<|user|>"])
         # this one is usually not in config.json anyway
         special_vocab._set_special_token("unk", tokenizer.get_added_vocab()["<|endoftext|>"])
+        # exclude glm-edge 1.5B & 4B
+        if self.hparams.get("partial_rotary_factor", 1.0) == 0.5:
+            print("add bos in model")
```

Comment:
A stray `print`? Also, `# exclude glm-edge 1.5B & 4B` is confusing to me as a comment, is it saying that in glm-edge 1.5B & 4B, there is no `partial_rotary_factor` so we don't put `[gMASK]` as bos token?

Is there a cleaner way to decide if to use `[gMASK]` as bos token? IMO might be good to just write something like `# GLM4 model has no bos token set in its tokenizer config, set it manually to [gMASK]` or something similar.

Reply:
The strategy I adopted is based on the fact that, as far as I know, both glm 4B and 1.5B are full rope and do not have bos.

Reply:
Additionally, I intend to incorporate your recommendations on the comments. I greatly appreciate your meticulous review.